_ingest-pipelines/processors/user-agent.md

```diff
@@ -0,0 +1,194 @@
+
+```json
+PUT _ingest/pipeline/user_agent_pipeline
+{
+  "description": "User agent pipeline",
+  "processors": [
+    {
+      "user_agent": {
+        "field": "user_agent",
+        "target_field": "user_agent_info"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/user_agent_pipeline/_simulate
+{
+  "pipeline": "user_agent_pipeline",
+  "docs": [
+    {
+      "_source": {
+        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
+          "user_agent_info": {
+            "name": "Chrome",
+            "original": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
+            "os": {
+              "name": "Windows",
+              "version": "10",
+              "full": "Windows 10"
+            },
+            "device": {
+              "name": "Other"
+            },
+            "version": "58.0.3029.110"
+          }
+        },
+        "_ingest": {
+          "timestamp": "2024-04-25T21:41:28.744407425Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+### Step 3: Ingest a document 
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=user_agent_pipeline
+{
+  "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36"
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The request indexes the document into the index `testindex1` and will index all documents with the user agent string parsed into it components:
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 66,
+  "result": "updated",
+  "_shards": {
+    "total": 2,
+    "successful": 1,
+    "failed": 0
+  },
+  "_seq_no": 65,
+  "_primary_term": 47
+}
+```
+
+### Step 4 (Optional): Retrieve the document
+
+To retrieve the document, run the following query:
+
+```json
+GET testindex1/_doc/1
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The response includes the original `user_agent` field and the parsed `user_agent_info` field with the device, operating system, and browser information: 
```

Comment:
@gaobinlong Is this an accurate description?

Reply:
It looks good to me.